python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -651,7 +790,7 @@ def _get_attachment_path(attachment_entity, bundle):
         return os.path.join(directory_path, file_name)
 
     @staticmethod
-    def _get_thumbnail_path(url, bundle, directory_only=False):
+    def _get_thumbnail_path(url, bundle, directory_only=False, use_default_extension=True):
```

Comment:
This is confusing to me. What if we instead of adding `use_default_extension` simply changed the return value to be `(path, extension)` and in the case when the extension is known, it is returned? That seems more transparent to me.

Then we didn't have to pass this `use_default_extension` flag around everywhere.